modin/data_management/factories/factories.py

```diff
@@ -35,30 +35,30 @@
 Abstract {role} factory which allows to override the IO module easily.
 
 This class is responsible for dispatching calls of IO-functions to its
-actual engine-specific implementations.
+actual backend-specific implementations.
 
 Attributes
 ----------
 io_cls : BaseIO
-    IO module class of the underlying engine. The place to dispatch calls to.
+    IO module class of the underlying backend. The place to dispatch calls to.
```

Comment:
this feels confusing a little bit, as `backend` in "Modin speech" is `Pandas` or `Pyarrow`... dunno how to fix it, though.

@devin-petersohn any ideas?

Reply:
There is an ambiguity between "backend" and "engine" terms for now. I think the change from "engine" to "backend" makes sense here. I hope that ambiguity will gone away after #2957 is resolved.